### PR TITLE
Added curl to the iceberg rest fixture Docker image #11688

### LINKED ISSUE #11688: Add `curl` to the `iceberg-rest-fixture` docker container
### Feature Request / Improvement

While checking the integration tests on Iceberg-rust (https://github.com/apache/iceberg-rust/pull/748 specifically), it might be nice to add curl to the docker image to enable the health check. With this health check, we no longer have to add logic (we have a loop in Iceberg-rust and a 10-second sleep in PyIceberg) to check if the container is alive and kicking.

### Query engine

None

### Willingness to contribute

- [ ] I can contribute this improvement/feature independently
- [ ] I would be willing to contribute this improvement/feature with guidance from the Iceberg community
- [ ] I cannot contribute this improvement/feature at this time

### PR DESCRIPTION
**Pull request by dominikhei**
Closes #11688

I have added curl to the docker image. If you want me to adjust the contribution in any way, e.g to also clean up the package cache to save space, let me know.

### CODE REVIEW

**Comment by Fokko** [comment]
@dominikhei Thanks for working on this. As you already suggested, it would be good to also clean up afterward, see best practices of Docker: https://docs.docker.com/build/building/best-practices/#apt-get

**Comment by dominikhei** [comment]
@Fokko I updated the PR!

**Comment by danielcweeks** [comment]
@dominikhei if we're going to add curl, shouldn't we setup the healthcheck as well?  It seems like it would make sense to tie this addition with its usage.

**Comment by dominikhei** [comment]
@danielcweeks let me know if it is fine if I just probe the `$REST_PORT `.

File: docker/iceberg-rest-fixture/Dockerfile [hunk 1]
```diff
@@ -36,6 +39,10 @@ ENV CATALOG_JDBC_USER=user
 ENV CATALOG_JDBC_PASSWORD=password
 ENV REST_PORT=8181
 
+# Healthcheck for the iceberg rest service
+HEALTHCHECK --start-period=30s --retries=3 \
+  CMD curl --fail http://localhost:$REST_PORT || exit 1
```
Comment (Fokko) [inline docker/iceberg-rest-fixture/Dockerfile:44]: I know that the JVM has a bit of a slow start, but 30 seconds seems a bit excessive for me. I would just start probing right away, and do this every second, for up to 10 seconds.
```suggestion
HEALTHCHECK --retries=10 --interval=1s \
  CMD curl --fail http://localhost:$REST_PORT/v1/config || exit 1
```

Reference: https://docs.docker.com/reference/dockerfile/#healthcheck

Checked this locally:
```
➜  iceberg-python git:(fd-use-load-catalog) curl --fail http://localhost:8181      
curl: (22) The requested URL returned error: 400
➜  iceberg-python git:(fd-use-load-catalog) curl --fail http://localhost:8181/v1/config
{"defaults":{},"overrides":{}}%                 
```

**Comment by dominikhei** [comment]
@Fokko Sorry, I did not see your message so far, thank you for the detailed explanation. I there anything left to do?

**Review by Fokko — APPROVED** [review]
@dominikhei No, this all looks good, thanks for working on this @dominikhei 🙌

**Pull request merged**